src/celeritas/global/CoreTrackView.hh

```diff
@@ -99,6 +100,8 @@ CoreTrackView::CoreTrackView(ParamsRef const& params,
     : states_(states), params_(params), thread_(thread)
 {
     CELER_EXPECT(thread_ < states_.size());
+    //TODO: Smarter mapping of thread to track index
+    states_.track_slots[thread_] = TrackSlotId{thread_.get()};
```

Comment:
Instead of doing this every time the core track view is instantiated (at least once per kernel), you should fill the TrackSlotIds during their initialization (the `resize` function). See the `resize` in `celeritas/track/TrackInitData.hh` or `celeritas/random/XorwowRngData.cc` for examples: you might define a helper function in a .cu file that hides `thrust::sequence` `https://thrust.github.io/doc/group__transformations_ga9d17b70466df6b9a57f94170935a03f8.html` and one in a .cc  (for host states) that uses `std::iota` to fill the data.

Reply:
I was working on that yesterday, I have a `sort_tracks` function in `TrackInitAlgorithms.hh` with different host / device implementation. It was failing because I was calling it in `ActionSequence::execute`. Can't quite use `std::iota` because the value type requires to be incrementable. I suppose we can implement `OpaqueID::operator++` as there is already support for addition.